www/activities/diary/js/group.js

```diff
@@ -130,25 +130,29 @@ app.Group = {
     });
 
     let categoryMacroNutriments = app.Settings.get("diary", "show-macro-nutriments-summary") ? [
-      this.getMacroNutrimentFooterText('F', nutrition.fat, categoryEnergyTotal),
-      this.getMacroNutrimentFooterText('C', nutrition.carbohydrates, categoryEnergyTotal),
-      this.getMacroNutrimentFooterText('P', nutrition.proteins, categoryEnergyTotal)
-    ].filter(text => text !== '') : [];
+      this.getMacroNutrimentFooterText('fat', nutrition.fat, categoryEnergyTotal),
+      this.getMacroNutrimentFooterText('carbohydrates', nutrition.carbohydrates, categoryEnergyTotal),
+      this.getMacroNutrimentFooterText('proteins', nutrition.proteins, categoryEnergyTotal)
+    ].filter(text => text !== null) : [];
 
     right.innerText = categoryMacroNutriments.join(' / ') +
         (categoryMacroNutriments.length > 0 ? ' / ' : '') +
         app.Utils.tidyNumber(Math.round(categoryEnergyTotal), energyUnitSymbol);
     row.appendChild(right);
   },
 
-  getMacroNutrimentFooterText: function(prefix/*: string*/, grams/*: number | undefined*/, categoryEnergyTotal/*: number*/) {
+  getMacroNutrimentFooterText: function (nutriment/*: string*/, grams/*: number | undefined*/, categoryEnergyTotal/*: number*/) {
     if (categoryEnergyTotal === 0) {
-      return '';
+      return null;
+    }
+    if (!app.Goals.showInDiary(nutriment)) {
+      return null;
     }
     if (grams === undefined || isNaN(grams)) {
-      return '';
-    }  else {
-      return prefix  + app.Utils.tidyNumber(Math.round(grams));
+      return null;
+    } else {
+      const nutrimentPrefix = nutriment.charAt(0).toUpperCase();
```

Comment:
Please make it possible to translate the prefixes into other languages. For example, carbohydrates are called "Kohlenhydrate" in German, so it should display the letter K in that case.

To do this, simply add the English abbreviations to the [`locale-en.json`](https://github.com/davidhealey/waistline/blob/master/www/assets/locales/locale-en.json) file, for example like this:

```diff
 {
   "nutriments": {
     ...
   },
+  "nutriments-abbreviations": {
+    "fat": "F",
+    "carbohydrates": "C",
+    "proteins": "P"
+  },
   "unit-symbols": {
     ...
   },
 }
```

Abbreviations in other languages can then be added later via Crowdin.

In the code, you can then get the localized prefix via the `app.strings` object:

```js
const nutrimentPrefix = app.strings["nutriments-abbreviations"][nutriment] || nutriment.charAt(0).toUpperCase();
```

Also, I think it might look better if instead of
`F9 / C13 / P6 / 150 kcal`
we display it as
`F 9 / C 13 / P 6 / 150 kcal`
or maybe even
`9 F / 13 C / 6 P / 150 kcal`

At least it would be worth exploring some of these alternatives...

Reply:
Also note that there could be languages where just using the first letter is not enough because the words for fat, carbs, and/or protein start with the same letter, so longer abbreviations are needed to distinguish them. Using the locale file and Crowdin makes that easy for us.

Reply:
Done